From d510aaaa7d5c7ca9a00fd105a34f394a30bd6192 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 14 Jul 2011 23:23:03 +0800
Subject: [PATCH 146/256] ENGR00153104 vpu: fix build failure on android
 environment

1) Fix build failure on android environment
2) Change TEMP_BUF_SIZE definition that was missed to update.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/sw_gbu.h   | 37 ++++++++++---------------------------
 vpu/vpu_reg.h  |  2 +-
 vpu/vpu_util.c |  1 -
 vpu/vpu_util.h | 12 ++----------
 4 files changed, 13 insertions(+), 39 deletions(-)

diff --git a/vpu/sw_gbu.h b/vpu/sw_gbu.h
index 8e2b5af..6a2f8c5 100644
--- a/vpu/sw_gbu.h
+++ b/vpu/sw_gbu.h
@@ -27,22 +27,8 @@
 #ifndef AVCODEC_GET_BITS_H
 #define AVCODEC_GET_BITS_H
 
-typedef unsigned char uint8_t;
-//typedef char int8_t;
-typedef unsigned int uint32_t;
-typedef int int32_t;
-typedef unsigned short uint16_t;
-typedef short int16_t;
-#if defined(_MSC_VER)
-typedef unsigned _int64 uint64_t;
-typedef _int64 int64_t;
-#else
-typedef unsigned long long uint64_t;
-typedef long long int64_t;
-#endif
-typedef unsigned char BYTE;
-
 #include <stdlib.h>
+#include <stdint.h>
 
 #if defined(_MSC_VER)
 #	define inline _inline
@@ -72,20 +58,17 @@ static inline const int sign_extend(int val, unsigned bits)
 }
 #endif
 
-#if defined(ALT_BITSTREAM_READER_LE) && !defined(ALT_BITSTREAM_READER)
-#   define ALT_BITSTREAM_READER
-#endif
-
-#if !defined(LIBMPEG2_BITSTREAM_READER) && !defined(A32_BITSTREAM_READER) && !defined(ALT_BITSTREAM_READER)
-#   if ARCH_ARM && !HAVE_FAST_UNALIGNED
-#       define A32_BITSTREAM_READER
-#   else
-#       define ALT_BITSTREAM_READER
-#   endif
-#endif
-
+#define ALT_BITSTREAM_READER
 #define VLC_TYPE int16_t
 
+/* bit input */
+/* buffer, buffer_end and size_in_bits must be present and used by every reader */
+typedef struct {
+    const Uint8 *buffer, *buffer_end;
+    int index;
+    int size_in_bits;
+} GetBitContext;
+
 typedef struct VLC {
 	int bits;
 	 VLC_TYPE(*table)[2];	///< code, bits
diff --git a/vpu/vpu_reg.h b/vpu/vpu_reg.h
index 8910e6e..999dd94 100644
--- a/vpu/vpu_reg.h
+++ b/vpu/vpu_reg.h
@@ -380,7 +380,7 @@
 #elif defined(IMX6Q)
 	#define CODE_BUF_SIZE                   (240 * 1024) /* define max is 240k byte currently */
 	#define FMO_SLICE_SAVE_BUF_SIZE         (32)
-	#define TEMP_BUF_SIZE                   (512 * 1024)
+	#define TEMP_BUF_SIZE                   (512 * 1024) + (FMO_SLICE_SAVE_BUF_SIZE * 1024 * 8)
 	#define PARA_BUF2_SIZE                  (2 * 1024)
 	#define PARA_BUF_SIZE                   (10 * 1024)
 #else
diff --git a/vpu/vpu_util.c b/vpu/vpu_util.c
index 42ae06a..8b96055 100644
--- a/vpu/vpu_util.c
+++ b/vpu/vpu_util.c
@@ -26,7 +26,6 @@
 #include "vpu_util.h"
 #include "vpu_io.h"
 #include "vpu_debug.h"
-#include "sw_gbu.h"
 
 #define MAX_VSIZE       8192
 #define MAX_HSIZE       8192
diff --git a/vpu/vpu_util.h b/vpu/vpu_util.h
index 48a6a2f..3aae860 100644
--- a/vpu/vpu_util.h
+++ b/vpu/vpu_util.h
@@ -23,7 +23,7 @@
 #include "vpu_reg.h"
 #include "vpu_lib.h"
 #include "vpu_io.h"
-//#include "sw_gbu.h"
+#include "sw_gbu.h"
 
 #define MAX_FW_BINARY_LEN		200 * 1024
 #define MAX_NUM_INSTANCE		8
@@ -298,16 +298,8 @@ typedef struct {
 
 } EncInfo;
 
-/* bit input */
-/* buffer, buffer_end and size_in_bits must be present and used by every reader */
 typedef struct {
-    const Uint8 *buffer, *buffer_end;
-    int index;
-    int size_in_bits;
-} GetBitContext;
-
-typedef struct {
-    // for Nieuport
+    /* for Nieuport */
     int picWidth;
     int picHeight;
     int alignedWidth;
-- 
1.8.0

